src/Components/CreateVideo/CreateVideo.tsx

```diff
@@ -1,11 +1,53 @@
-import React, {useEffect, useState} from 'react';
+import React, { useState } from "react";
+import { useInterval } from "beautiful-react-hooks";
+import { Button, Modal } from "react-bootstrap";
+import { Media } from "../";
+import styles from "./CreateVideo.module.css";
 
-import styles from './CreateVideo.module.css';
-
-const CreateVideo: React.FC = () => {
-    return(
-        <div>create</div>
-    )
-}
-
-export default CreateVideo;
\ No newline at end of file
+type Props = {
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  show: boolean;
+  medias: Media[];
+};
+export const CreateVideo: React.FC<Props> = ({ setShow, show, medias }) => {
+  const [mediaCounter, setMediaCounter] = useState<number>(0);
+  const [imageDuration, setImageDuration] = useState<number>(5000);
+  const changeImage: () => void = () => {
+    if (medias == null) {
+      return;
+    }
+    if (mediaCounter >= medias.length) {
+      // clearing interval for media switching within the video
+      clearInterval();
+    } else {
+      // video ended
+      setMediaCounter((mediaCounter) => mediaCounter + 1);
+    }
+  };
+  const [isCleared, clearInterval] = useInterval(changeImage, imageDuration);
```

Comment:
Can this be above the `changeImage` declaration? Otherwise it's hard to read where `clearInterval` comes from